types/firestore.d.ts

```diff
@@ -1000,6 +1065,29 @@ declare namespace FirebaseFirestore {
      * @return true if this `CollectionReference` is equal to the provided one.
      */
     isEqual(other: CollectionReference): boolean;
+
+    /**
+     * Retrieves all documents in this CollectionReference.  Documents with
+     * subcollections that contain no other data are returned as
+     * `DocumentSnapshots` whose `exists` property is set to false.
+     *
+     * @param options An object to configure the behavior of this `getAll` call
+     * (i.e. the set of fields to return).
+     * @return A Promise that will be resolved with the contents of of this
+     * collection.
+     */
+    getAll(options?:GetOptions): Promise<DocumentSnapshot>;
+
+    /*
+     * Retrieves all documents in this CollectionReference in a Node Stream.
+     * Documents with subcollections that contain no other data are returned as
+     * `DocumentSnapshots` whose `exists` property is set to false.
+     *
+     * @param options An object to configure the behavior of this `streamAll`
+     * call (i.e. the set of fields to return).
+     * @return A stream of DocumentSnapshots.
+     */
+    streamAll(options?:GetOptions): NodeJS.ReadableStream;
```

Comment:
Is this streaming all the way back to the backend?  So I could read this "slowly" (e.g. 1 document per second) and it won't download the entire list?  And I can stop at any time?  (I'm trying to figure out if we think this is a suitable replacement for pagination).